app/models/host_info_providers/puppet_info.rb

```diff
@@ -21,8 +21,23 @@ def puppetclass_parameters
       values = keys.values_hash(host)
 
       klasses = {}
+      klass_error = {}
```

Comment:
This could easily take advantage of WIP code which is emerging at https://github.com/theforeman/foreman/pull/5785/files

Reply:
O.o! That's perfect for this.